### PR TITLE
BE: Request Controllers

### PR DESCRIPTION
**Pull request by dproc96**
**What it does**
- Create GET /requests/me route to get all of the user's requests
- Create POST /requests/request route to create a new request
- Create PUT /requests/request/:id rout to update a request
- Creates controllers for all three requests

**What to avoid**
- I wasn't sure how to implement testing for these routes as they require a logged in user. I'm planning to create a new branch and implement testing this weekend but figured it was better to move on to other priorities for now with the presentation on Friday

### CODE REVIEW

File: server/controllers/request.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ const getRequestsByUser = (req, res, next) => {
     return next(createError(403));
```
Comment (Braagaa) [inline server/controllers/request.js]: Just a minor note that you don't have to check for req.user. Passport will automatically throw a 403 Forbidden if user is unauthenticated.

Comment (Braagaa) [reply server/controllers/request.js]: And thanks for prettifying my code lol

**Comment by dproc96** [comment]
Added test cases and verified they were working!

File: server/controllers/request.js (hunk 1 above)
Comment (dahrk) [reply server/controllers/request.js]: @dproc96 small change for you before merging. Thanks Michael!

File: server/test/requestRoutes.spec.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+              .end(() => {
+                done();
+              });
+          });
+        });
+      });
+    });
+  });
+  let requestId;
+  it("POST /request/create Creates a request with a correct POST request", done => {
```
Comment (dahrk) [inline server/test/requestRoutes.spec.js:74]: might be good to update this test with the user_id somehow and make sure it's symmetric?

**Review by dahrk — APPROVED** [review]
Code looks great, take a look at Michael's comment and maybe think about adding that extra test coverage before merging.

**Pull request merged**